### PR TITLE
[react-native-router] added missing properties to NativeRouter and fixed Link component type

### PR DESCRIPTION
**Pull request by AlmirKadric**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by AlmirKadric** [comment]
There is one other thing I would like to add to this PR, however the linter breaks when I do. I would like to add an any union to the `Link` properties. The reason for this is due to the face that the Link component will proxy any properties you give it to the component you feed in. This means you can quite literally give it anything.

I can either remove the LinkProps object and replace it with any, but we lose auto completion for known properties, or I can create a union against LinkProps. Please advise on best way to move forward.

File: types/react-router-native/index.d.ts [hunk 1]
```diff
@@ -31,15 +31,24 @@ export class AndroidBackButton extends React.Component<AndroidBackButtonProps> {
...
   replace?: boolean;
   style?: any;
   to: H.LocationDescriptor;
 }
 
-export class Link extends React.Component<LinkProps> {}
+/**
+ * Disable the "no-any-union" rule here as the Link component can accept any property.
+ * It will feed them to the given component. However if we just use any we will lose
+ * autocomplete on known properties. So it is better to disable this rule in this one
```
Comment (DanielRosenwasser) [inline types/react-router-native/index.d.ts]: First off,

Comment (DanielRosenwasser) [inline types/react-router-native/index.d.ts]: First off

image

I'm not seeing any augmentation in completion. Second, if `LinkProps` can accept any property, you'd be better off adding an index signature like

```diff
   replace?: boolean;
   style?: any;
   to: H.LocationDescriptor;
+  [propName: string]: any;
 }
```

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]

File: types/react-router-native/index.d.ts (hunk 1 above)
Comment (AlmirKadric) [reply types/react-router-native/index.d.ts]: as seems i missed that inside the common mistake section, sorry about that will fix

**Review by DanielRosenwasser — APPROVED** [review]

**Pull request merged**